gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/JdbcProvider.java

```diff
@@ -21,18 +27,50 @@
  * @author nveeramr
  */
 public class JdbcProvider extends BasicDataSource {
+  private static final Logger LOG = LoggerFactory.getLogger(JdbcProvider.class);
+  private Tunnel tunnel;
+
   // If extract type is not provided then consider it as a default type
   public JdbcProvider(String driver, String connectionUrl, String user, String password, int numconn, int timeout) {
-    this.connect(driver, connectionUrl, user, password, numconn, timeout, "DEFAULT");
+    this.connect(driver, connectionUrl, user, password, numconn, timeout, "DEFAULT", null, -1);
   }
 
   public JdbcProvider(String driver, String connectionUrl, String user, String password, int numconn, int timeout,
       String type) {
-    this.connect(driver, connectionUrl, user, password, numconn, timeout, type);
+    this.connect(driver, connectionUrl, user, password, numconn, timeout, type, null, -1);
+  }
+
+  public JdbcProvider(String driver, String connectionUrl, String user, String password, int numconn, int timeout,
+      String type, String proxyHost, int proxyPort) {
+    this.connect(driver, connectionUrl, user, password, numconn, timeout, type, proxyHost, proxyPort);
   }
 
   public void connect(String driver, String connectionUrl, String user, String password, int numconn, int timeout,
-      String type) {
+      String type, String proxyHost, int proxyPort) {
+
+    if (proxyHost != null && proxyPort > 0) {
+      String remoteHost = "";
+      int remotePort = 0;
+      // TODO make connection Url parsing much more robust -- some connections URLs can have colons and slashes in the
+      // weirdest places
+      int hostStart = connectionUrl.indexOf("://") + 3;
+      int portStart =  connectionUrl.indexOf(":", hostStart);
+      remoteHost = connectionUrl.substring(hostStart, portStart);
+      remotePort = Integer.decode(connectionUrl.substring(portStart + 1, connectionUrl.indexOf("/", portStart)));
+
+      try {
+        tunnel = Tunnel.build(remoteHost, remotePort, proxyHost, proxyPort);
+        int tunnelPort = tunnel.getPort();
+        //mangle connectionUrl, replace hostname with localhost -- hopefully the hostname is not needed!!!
+        String newConnectionUrl = connectionUrl.replaceFirst(remoteHost, "127.0.0.1")
+            .replaceFirst(":" + remotePort, ":" + tunnelPort);
+        connectionUrl = newConnectionUrl;
+      } catch (IOException ioe) {
+        LOG.warn("Failed to open tunnel via proxy " + proxyHost + ":" + proxyPort
```

Comment:
This method should just `throw IOException` rather than catch the `IOException` and simply print a warning. This way the job will fail if the connection to the tunnel fails, which is better than simply printing a warning and allowing the job to continue.